pontoon/base/adapter.py

```diff
@@ -0,0 +1,30 @@
+from allauth.socialaccount.adapter import DefaultSocialAccountAdapter
+from allauth.socialaccount.providers.base import AuthProcess
+
+from allauth.socialaccount.helpers import _add_social_account
+from django.contrib.auth.models import User
+
+
+class PontoonSocialAdapter(DefaultSocialAccountAdapter):
+    """
+    It's required to merge old accounts created via django-browserid
+    with accounts created by django-allauth.
+    """
+    def pre_social_login(self, request, sociallogin):
+        """connect existing accounts with existing accounts."""
+        email = sociallogin.account.extra_data.get('email')
+
+        if not email:
+            return
+
+        try:
+            user = User.objects.get(email=email)
+        except User.DoesNotExist:
+            return
+
+        # This is a fallback for accounts provisioned during django-browserid.
+        # We assume that selected all emails from providers are verified.
```

Comment:
Yeah, I need to rephrase that.
Without this adapter, django-allauth couldn't automatically connect accounts from the old auth system and required manual intervention/conflict resolution from the user who is trying to log in.